app/util/variables.ts

```diff
@@ -204,4 +204,33 @@ export function getVariablesForCollection(
       versionId: collection.version_id, variables, coordinateVariables });
   }
   return variableInfo;
+}
+
+/**
+ * Validate that the extend query parameter is both a valid variable name or concept ID
+ * and falls within the requested variable(s) (OGC collectionId) and collection(s).
+ * Assumes that the extend variable only needs to be found within one of the requested collections.
+ * @param extendParam - The name or concept ID of the variable to be extended (extend query parameter)
```

Comment:
I tested it out and we're expecting only a single value for the extend parameter - that means we can't allow extending across both lat and lon for example. Is that fine?

Reply:
I think that's a use case that extend will likely want to support in the future.